core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java

```diff
@@ -192,6 +197,37 @@ public BigQuerySqlDialect(SqlDialect.Context context) {
     }
   }
 
+  /**
+   * For usage of TRIM, LTRIM and RTRIM in BQ see
+   * <a href="https://cloud.google.com/bigquery/docs/reference/standard-sql/functions-and-operators#trim">
+   *  BQ Trim Function</a>.
+   */
+  private void unparseTrim(SqlWriter writer, SqlCall call, int leftPrec,
+      int rightPrec) {
+    assert call.operand(0) instanceof SqlLiteral : call.operand(0);
```

Comment:
Also need to validate the second operand?

Reply:
@amaliujia I have remove this check as this is an unnessary check over here because in SqlTrimFunction class checkOperandTypes method perform the Operands check.
Removed this assert as the operands type check is already done in SqlTrimFunction class.

Reply:
Removed this assert as the operands type check is already done in SqlTrimFunction class